src/alexapi/device_platforms/magicmirror.py

```diff
@@ -0,0 +1,172 @@
+		if self._config['debug']:
+			print("Starting Magic Mirror platform HTTP Server")
+		self.serverthread.start()
+		# Start HB
+		if self._config['debug']:
+			print("Starting Magic Mirror heartbeat with " + str(self.HB_TIMER) + " second interval")
+		self.mm_heartbeat()
+
+	def indicate_recording(self, state=True):
+		if state:
+			if self._config['debug']:
+				print("Indicate Start Recording")
+			self.update_mm("recording")
+		else:
+			if self._config['debug']:
+				print("Indicate Stop Recording")
+			self.update_mm("idle")
+
+	def indicate_playback(self, state=True):
+		if state:
+			if self._config['debug']:
+				print("Indicate Start Playing")
+			self.update_mm("playback")
+		else:
+			if self._config['debug']:
+				print("Indicate Stop Playing")
+			self.update_mm("idle")
+
+	def indicate_processing(self, state=True):
+		if state:
+			if self._config['debug']:
+				print("Indicate Start Processing")
+			self.update_mm("processing")
+		else:
+			if self._config['debug']:
+				print("Indicate Stop Processing")
+			self.update_mm("idle")
+
+	def should_record(self):
+		if self.req_record:
+			self.req_record = False
+			return True
+		return False
+
+	def update_mm(self, status):
+		address = ("http://" + self.MM_HOST + ":" + self.MM_PORT + "/alexapi?action=AVSSTATUS&status=" + status)
+		try:
+			if self._config['debug']:
+				print("Calling URL: " + address)
+			response = urllib2.urlopen(address).read()
+			if self._config['debug']:
+				print("Response: " + response)
+		except Exception, e:
+			print(e)
+
+	def mm_heartbeat(self):
+		if self.shutdown: return
+		threading.Timer(self.HB_TIMER, self.mm_heartbeat).start()
+		address = ("http://" + self.MM_HOST + ":" + self.MM_PORT + "/alexapi?action=AVSHB")
+		try:
+			if self._config['debug']:
+				print("Sending MM Heatbeat")
+			response = urllib2.urlopen(address).read()
+		except Exception, e:
+			print(e)
+
+	def HTTPcallback(self, query_dict):
+		if (query_dict['action'][0] == "requestrecord"):
+			self.req_record = True
+			return True
+		else:
+			return False
+
+	def cleanup(self):
+		if self._config['debug']:
+			print("Cleaning up Magic Mirror platform")
+		self.httpd.shutdown()
+		self.shutdown = True
+
+
+# Subclass HTTPServer with additional callback
+class CallbackHTTPServer(BaseHTTPServer.HTTPServer):
+	def set_callback(self, callback):
+		self.RequestHandlerClass.set_callback(callback)
+
+# Subclass Request Handler to use callback
+class MMHTTPHandler(BaseHTTPServer.BaseHTTPRequestHandler):
+	@classmethod
+	def set_callback(self, callback):
+		self.callback = callback
+
+	def do_HEAD(self):
+		self.send_response(200)
+		self.send_header("Content-type", "text/html")
+		self.end_headers()
+
+	def do_GET(self):
+		self.send_response(200)
+		self.end_headers()
+
+		query = urlparse.urlsplit(self.path).query
+		query_dict = urlparse.parse_qs(query)
+
+		if 'action' in query_dict.keys():
+			if (self.callback(query_dict)):
+				self.wfile.write("{\"status\":\"success\"}")
```

Comment:
Does ' work in python like in PHP so you don't have to escape the crap out of it?